src/averageAges.js

```diff
@@ -55,7 +65,26 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(child =>
+      child.sex === 'm' && people.some(mother => mother.name === child.mother)
+    )
+    : people.filter(child =>
+      people.some(mother => mother.name === child.mother)
+    );
+
+  children.map(child => {
```

Comment:
```suggestion
  children.forEach(child => {
```

Reply:
Method `map` returns a new array, but you don't need the new array, you mutate the existing array